discovery.go

```diff
@@ -167,6 +171,9 @@ func (r *Router) DiscoveryAllBuckets(ctx context.Context) error {
 	errGr, ctx := errgroup.WithContext(ctx)
 
 	idToReplicasetRef := r.getIDToReplicaset()
+	if idToReplicasetRef == nil {
```

Comment:
I think this check is redundant therefore should be removed, because it should never happen.

P.S. TestRouter_DiscoveryAllBuckets also should be removed, because depends on this error